cmd/utils/nodecmd/util.go

```diff
@@ -0,0 +1,237 @@
+// Modifications Copyright 2023 The klaytn Authors
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package nodecmd
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/klaytn/klaytn/accounts/keystore"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/cmd/utils"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/crypto/sha3"
+	"github.com/klaytn/klaytn/governance"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/rlp"
+	"gopkg.in/urfave/cli.v1"
+)
+
+const pgmInput = `
+<extra> <header file (json format)>
+<vote>  <bytes>
+<gov>   <bytes>
+<key>   <keystore path> <password>
+`
+
+var ErrInvalidParam = errors.New("Invalid length of parameter")
+
+var UtilCommand = cli.Command{
+	Action:    utils.MigrateFlags(parse),
+	Name:      "util",
+	Usage:     "offline utility " + pgmInput,
+	ArgsUsage: " ",
+	Category:  "MISCELLANEOUS COMMANDS",
+}
+
+func hex2Bytes(s string) []byte {
+	if data, err := hexutil.Decode(s); err == nil {
+		return data
+	} else {
+		panic(err)
+	}
+}
+
+func printUsage() error {
+	fmt.Println("Usage: ./kxn util " + pgmInput)
+	return ErrInvalidParam
+}
+
+func validateInput(ctx *cli.Context, parseTyp string) error {
+	switch parseTyp {
+	case "vote", "extra", "gov":
+		if len(ctx.Args()) != 2 {
+			return printUsage()
+		}
+	case "key":
+		if len(ctx.Args()) != 3 {
+			return printUsage()
+		}
+	}
+	return nil
+}
+
+func parse(ctx *cli.Context) error {
+	parseTyp, m := ctx.Args().Get(0), make(map[string]interface{})
```

Comment:
Can we make "m" variable inside each function? It seems odd that parse function has to take an empty map.

```go
func parseVote(bytes []byte) (map[string]interface{}, error) {
    m := make(map[string]interface{})
    ...
    return m, nil
}
```